corehq/ex-submodules/casexml/apps/case/mock/case_block.py

```diff
@@ -98,8 +100,13 @@ def _to_json(self):
             'update': self.update,
         }
 
+        external_id = self.external_id
+        # This is probably not the best way, but is needed currently.
+        if self.domain and USE_CUSTOM_EXTERNAL_ID_CASE_PROPERTY.enabled(self.domain):
+            external_id = self.update.get('external_id', external_id)
+
         result['update'].update({
-            'external_id': self.external_id,
```

Comment:
Isn't this just a test-code utility, don't we need to apply this to the actual case-rule code as well?

Reply:
[Here's](https://github.com/dimagi/commcare-hq/blob/a28b63a95052d08c5dc079e37c9c0e3d7026154e/corehq/apps/hqcase/utils.py#L136) where we actually use this outside of just the test code. I'm not sure what you refer to when you say "actual case-rule code" (seeing as this is the path the actual code follows)? 

I do think the fact that it sits inside the `mock` folder is confusing...